src/IO/WriteBufferFromPocoSocketChunked.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <Common/logger_useful.h>
+#include <IO/WriteBufferFromPocoSocket.h>
+
+
+namespace DB
+{
+
+class WriteBufferFromPocoSocketChunked: public WriteBufferFromPocoSocket
+{
+public:
+    explicit WriteBufferFromPocoSocketChunked(Poco::Net::Socket & socket_, size_t buf_size = DBMS_DEFAULT_BUFFER_SIZE) : WriteBufferFromPocoSocket(socket_, buf_size), log(getLogger("Protocol"))
+    {
+        chassert(buf_size <= std::numeric_limits<std::remove_reference_t<decltype(*chunk_size_ptr)>>::max() && buf_size > sizeof(*chunk_size_ptr));
```

Comment:
`chassert` won't do anything in the release build and won't prevent the problems. Exception or setting `buf_size` to `min(sizeof(*chunk_size_ptr) + 1, buf_size)` will be better imo